internal/datastore/proxy/schemacaching/watchingcache.go

```diff
@@ -89,8 +89,7 @@ func createWatchingCacheProxy(delegate datastore.Datastore, c cache.Cache[cache.
 
 		gcWindow:       gcWindow,
 		watchHeartbeat: watchHeartbeat,
-		closed:         make(chan bool, 2),
```

Comment:
FYI making this channel buffered means that the sender will _not_ block. So `Close()` isn't waiting until the receiver receive and the goroutines close. 

But, funnily enough, removing the buffer makes the tests never finish. So i'm replacing with a `waitgroup`.